src/Compilers/CSharp/Test/Semantic/Semantics/UsingStatementTests.cs

```diff
@@ -1452,6 +1453,280 @@ static void Main()
             CreateCompilation(source).VerifyDiagnostics();
         }
 
+        [Fact]
+        public void UsingPatternAsyncTest()
+        {
+            var source = @"
+namespace System
+{
+    public interface IAsyncDisposable
+    {
+        System.Threading.Tasks.ValueTask DisposeAsync();
+    }
```

Comment:
Is this interface definition needed? Consider adding a similar test without this.

Reply:
Yes. We check for the presence of the interface before we do any pattern lookup, to stay consistent with the priority order of IDiposable. (Interface -> instance method -> Extension method). That means we have to have the interface to be able to do await using() statements.

Reply:
Thanks. Consider adding a test without the interface definition (if there isn't one already) so it's clear that the interface is necessary.

---
In reply to: [235181492](https://github.com/dotnet/roslyn/pull/31201#discussion_r235181492) [](ancestors = 235181492)